drift/lib/src/runtime/types/sql_types.dart

```diff
@@ -168,7 +168,8 @@ class BlobType extends SqlType<Uint8List> {
 
   @override
   Uint8List? mapFromDatabaseResponse(dynamic response) {
-    return response as Uint8List?;
+    var list = response.codeUnits;
+    return Uint8List.fromList(list);
```

Comment:
`response` should really be a `Uint8List` already, not a string. Now this method only works when given a string, but we can actually store blobs in the database.

Reply:
Correct. To make it more robust, how about:
```
  @override
  Uint8List? mapFromDatabaseResponse(dynamic response) {
    if (response is String) {
      var list = response.codeUnits;
      return Uint8List.fromList(list);
    }
    return response as Uint8List?;
  }
```